app/requirements/basic.py

```diff
@@ -0,0 +1,33 @@
+
+
+class Requirement():
+
+    def __init__(self, requirement_info):
+        self.enforcements = requirement_info['enforcements']
+        self.fact_relationships = requirement_info['fact_relationships']
+
+    def enforce(self, potential_fact, used_facts, all_operation_facts):
+        """
+        Given a potential fact, all facts used by the current link and all operation facts, determine if it complies
+        with this fact relationships enforcement mechanism
+        :param potential_fact: List of facts that are available
+        :param used_facts
+        :param all_operation_facts
+        :return: True if it complies, False if it doesn't
+        """
+        for uf in used_facts:
+            f = self._get_fact(all_operation_facts, uf)
+            if self.enforcements.get('source') == f.get('property'):
```

Comment:
now the big question: can you make this for-loop simpler? :)

Reply:
one question to help prompt you: do you see any of these loops/if-checks as being "generic" enough for future Requirement classes? for example, do you see any way to leverage a BaseRequirement class, similar to how we have a BaseParser class?

Reply:
I'll see what I can do